scripts/release/build/update-package-versions.js

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const chalk = require('chalk');
+const {exec} = require('child-process-promise');
+const {readFileSync, writeFileSync} = require('fs');
+const {readJson, writeJson} = require('fs-extra');
+const {join} = require('path');
+const {projects} = require('../config');
+const {execUnlessDry, logPromise} = require('../utils');
+
+const update = async ({cwd, dry, version}) => {
+  try {
+    // Update root package.json
+    const packagePath = join(cwd, 'package.json');
+    const rootPackage = await readJson(packagePath);
+    rootPackage.version = version;
+    await writeJson(packagePath, rootPackage, {spaces: 2});
+
+    // Update ReactVersion source file
+    const reactVersionPath = join(cwd, 'src/ReactVersion.js');
+    const reactVersion = readFileSync(reactVersionPath, 'utf8').replace(
+      /module\.exports = '[^']+';/,
+      `module.exports = '${version}';`
+    );
+    writeFileSync(reactVersionPath, reactVersion);
+
+    // Update renderer versions and peer dependencies
+    const updateProjectPackage = async project => {
+      const path = join(cwd, 'packages', project, 'package.json');
+      const json = await readJson(path);
+      json.version = version;
```

Comment:
I _think_ `react-reconciler` should be special cased here to not have lock-step versioning (though not sure what to do besides just give it minor bumps)

Reply:
Hm. Why would we want to release reconciler independently? In the past, we've released react-* things in lock step (even though often not strictly required) b'c it's easier for external people to see and reason about.

Reply:
🤔  Since the API isn’t semver stable yet keeping it sub 1.x avoids issues for breaking the likely breaking API changes that will take place as the async work makes progress.

Reply:
Gotcha.

I see 2 potential downsides for this change:

1: It adds complexity to an automated process. (I'm not saying this a compelling argument _not_ to do it but it is a downside.)

We could require a _separate_ argument for the reconciler version- or we could require the release engineer to manually update it before hand but both are a worse experience for the release engineer. Maybe we should just handle the case of a version < 1 by auto-incrementing (eg `0.1.0` is less than `1.0.0` so we special case to auto-bump to `0.1.1`). What are your thoughts?

2: More importantly, I worry it adds confusion for end-users. It's a nicer developer experience to know that `react-reconciler` version X is compatible with `react` version X without having to rely on the (typically noisy, often overlooked) peer dependency warning mechanism.

I think this is more important and we should _eventually_ aim to synchronize the version of `react-reconciler` with `react`. (At least once the API stabilizes.)

Reply:
We definitely **should not** be versioning the reconciler package together with other packages right now. I'd be okay bumping a minor on every release (e.g. `0.1.0` -> `0.2.0` -> `0.3.0`). It is changing *way* too fast and I don't see it stabilizing for another six months or so.

Reply:
Releasing patch by default won't work too. We will break it between main React patches pretty often. Ideally we should increment a `0.x` minor for it when the Flow file changes, but I think the low-maintenance “always increment” approach will work fine in the meantime. People using this know it’s very experimental anyway.

Reply:
>I worry it adds confusion for end-users

There are, like, 10 end users. Only people creating custom renderers are end users, and they are already used to patching React and digging up super complicated internals just to get it working. So this is already a large improvement for them.

Reply:
> We definitely **should not** be versioning the reconciler package together with other packages right now.

Yes, that's become apparent 😄  I didn't realize the degree of expected instability for this package because I haven't been following it closely.

> but I think the low-maintenance “always increment” approach will work fine in the meantime.

Great.

> There are, like, 10 end users.

That's a good thing to be reminded of.

> There shouldn't be an issue with react <-> react-reconciler dependencies and we can safely say any react-reconciler version is friends with react@16. This is because we don’t expose “important” things from react package anymore.

I'm skeptical of this 😄  but so long as this is a temporary thing (unlocked versions) until the reconciler package stabilizes, I'm not opposed.